src/IR/IR.ts

```diff
@@ -99,6 +104,7 @@ export type Node =
   | MutatingInfix
   | IndexCall
   | RangeIndexCall
+  | ScanningMacroCall
```

Comment:
New node kinds need to be added to parser, polygolf emitter and to parse-emit.test.ts.